### DISCUSSION TITLE
How to lazy load JSON translation file without adding Namespace?

### DISCUSSION THREAD

**Discussion by kylsgl** [i18next/i18next#2205] [Q&A] [answered]
Currently I'm loading the JSON translation files directly to the resource obj.
```js
const opts = {
	debug: true,
	fallbackLng: 'en',
	lng: getLanguage(),
	resources: {
		en,
		es,
	},
}
```
I can then access the translation like this:
```js
i18n.t('subhead', { ns: 'app' })
```

However, when I try to lazy load the translation files with this:
```js
const lazy: Module = {
	init() {
		// no-op
	},
	read(
		language: string,
		_namespace: string,
		callback: (...args: unknown[]) => unknown,
	) {
		import(`./assets/locales/${language}.json`)
			.then((data): unknown => callback(null, data?.default ?? data))
			.catch(callback);
	},
	type: 'backend',
} 
```
It adds a **translation** namespace:
```js
i18next::backendConnector: loaded namespace translation for language en 
```
And now I have to use **translation** as **ns** to access the translation
```js
i18n.t('app.subhead', { ns: 'translation' })
```

How can I prevent this behavior so I can keep using "app" as the namespace?
```js
i18n.t('subhead', { ns: 'app' })
```

**Comment by jamuhl** [comment, marked as answer]
set `defaultNS`in init options from translation to app https://www.i18next.com/overview/configuration-options#languages-namespaces-resources

**Reply by jamuhl** [reply]
also make sure your JSON is correct -> keys directly on first level...not below a ns `{ key1, key2 }` not `{ app: { key1, key2 } }`

**Reply by kylsgl** [reply]
But that means I will have to use the "app" namespace every time?


This is my translation JSON structure:

```json
{
	"app":{
		"subhead":"Hello",
	},
	"components": {
		"modal": {
			"close": "Close modal",
		}
	},
	"page": {
		"home": "Home",
		"search": "Search"
	},
}
```
And access it like this
```js
i18n.t('subhead', { ns: 'app' })
i18n.t('modal.close', { ns: 'components' })
i18n.t('home', { ns: 'page' })
```

**Reply by jamuhl** [reply]
namespaces are files...you can use https://www.i18next.com/overview/api#getfixedt keyprefix

**Reply by jamuhl** [reply]
or adapt https://github.com/i18next/i18next-multiload-backend-adapter

**Reply by kylsgl** [reply]
> namespaces are files...you can use https://www.i18next.com/overview/api#getfixedt keyprefix

I see. I have been using it wrong. Thanks.
